### PR TITLE
flux-kvs(1): improve man page formatting

### PR DESCRIPTION
**Pull request by garlick**
Problem: flux-kvs(1) is hard to use as a reference.

Improve the synopsis, description, and break out sub-commands into sub-sections with properly declared options.

### CODE REVIEW

File: doc/man1/flux-kvs.rst [hunk 1]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+
+.. option:: -u, --uniq
+
+  With :option:`--watch`, suppress output when value is the same, even
+  if the watched key was the target of a KVS commit.
+
+.. option:: -A, --append
+
+  With :option:`--watch`, display only the new data when the watched
+  is appended to.
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:131]: "when the watched **value** is appended to"?

File: doc/man1/flux-kvs.rst [hunk 2]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+  replaced.  The :option:`--full` option ensures these changes are reported
+  as well, at greater overhead.
+
+put
+---
+
+.. program:: flux kvs put
+
+Set *key* to *value*.  If *key* exists, the current value is overwritten.
+If multiple *key=value* pairs are specified, they are send as one
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:148]: s/send/sent/

File: doc/man1/flux-kvs.rst [hunk 3]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+If multiple *key* arguments are specified, their values are concatenated.
+A newline is appended to each value, unless :option:`--raw` is specified or
+the value is zero length.
+
+
+.. option:: -h, --help
+
+  Display help on this sub-command.
+
+.. option:: -N, --namespace=NAME
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:88]: personal preference ... `--help` option feels the least useful in the manpage (i.e. you're already in the manpage).  List it last amongst all the options?

File: doc/man1/flux-kvs.rst [hunk 4]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+.. option:: -n, --no-merge
+
+   Set the ``NO_MERGE`` flag on the commit to ensure it is not combined with
+   other commits.  The KVS normally combines contemporaneous commits to save
+   work, but since commits succeed or fail atomically, this a commit could
+   fail due to collateral damage.  This option prevents that from happening.
+
+.. option:: -A, --append
+
+   Append value(s) to key instead of overwriting.
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:195]: no `(s)`?  Or should be `key(s)`?  I'd leave it singular.

File: doc/man1/flux-kvs.rst [hunk 5]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+   After the commit has completed, display the new root sequence number
+   or "version".
+
+readlink
+---------
+
+.. program:: flux kvs readlink
+
+Print the symbolic link target of *key*.  The target may be another key name,
+or a :option:`namespace::key` tuple.
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:362]: similar to other sections above, say it's an error if `key` is not a link?

File: doc/man1/flux-kvs.rst [hunk 6]
```diff
@@ -65,170 +65,616 @@ setting the namespace in the environment variable :envvar:`FLUX_KVS_NAMESPACE`.
...
+namespace remove
+----------------
+
+.. program:: flux kvs namespace remove
+
+Remove a KVS namespace.
+
+.. option:: -h, --help
+
+  Display help on this sub-command.
```
Comment (chu11) [inline doc/man1/flux-kvs.rst:573]: perhaps note that remove is done in the background? i.e. may not be removed when command returns.

**Review by chu11 — COMMENTED** [review]
overall looks good, just some small comments

File: doc/man1/flux-kvs.rst (hunk 3 above)
Comment (garlick) [reply doc/man1/flux-kvs.rst:88]: Great point. Instead of repeating that option and the namespace one, I'll just mention it in the section intro.

**Comment by garlick** [comment]
Thanks @chu11, that was super helpful.  I fixed those things and also reworded the last paragraph of the DESCRIPTION about namespaces, and added RFC 18 to the FLUX RFC section.

**Review by chu11 — APPROVED** [review]
LGTM

**Pull request merged**